src/cypress/utils/navigation.ts

```diff
@@ -8,29 +8,80 @@ import { resolveIdentifier } from './index'
 interface NavigationOptions {
   categoryId: string
   random: boolean
+  pageIndex?: number
 }
 
-export function goToSearchPage(options?: NavigationOptions) {
-  if (!options) return
+export function goToSearchPage(options?: NavigationOptions): boolean {
+  if (!options) {
+    return false
+  }
 
   if (options.categoryId) {
     cy.visit(options.categoryId)
+
+    return true
   }
 
   if (options.random) {
     cy.visit('/')
 
     cy.get('[data-testid="categoryLink"]')
       .its('length')
-      .then((itemsLength) => {
-        const itemIndex = Math.floor(Math.random() * itemsLength)
+      .then(($length) => {
+        const itemIndex = Math.floor(Math.random() * $length)
 
         cy.get(`[data-testid="categoryLink"]`)
           .eq(itemIndex)
-          .invoke('show')
-          .click({ force: true })
+          .then(($link) => {
+            const url = new URL($link.prop('href'))
+
+            cy.intercept('GET', `/page-data/${url.pathname}/page-data.json`).as(
+              `pageLoad${url.pathname}`
+            )
+            cy.get(`[data-testid="categoryLink"]`)
+              .eq(itemIndex)
+              .click({ force: true })
```

Comment:
This parameter `force: true` is necessary? I'm questioning because I saw that you removed this from other tests.

Reply:
It helps the test run smoothly in case the link isn't visible (eg. it's in a dropdown menu)